src/steamship/data/block.py

```diff
@@ -250,6 +250,17 @@ def raw_data_url(self) -> Optional[str]:
         else:
             return None
 
+    @property
+    def response_sent(self) -> bool:
+        return get_tag_value_key(
+            self.tags, TagValueKey.BOOL_VALUE, kind=DocTag.CHAT, name=ChatTag.RESPONSE_SENT
+        )
+
+    def set_response_sent(self, is_sent: bool):
```

Comment:
is there any value in recording any of the following in the tag:
- recording date/time of response
- recording block IDs / hash of response 
- context ID of response
- request ID that led to response

?

Reply:
This is a great idea -- and trying to add these in led to a worthwhile question to bounce back:

* Block IDs aren't yet available from within `run_agent` since we request that the webhook implementor be the one to decide whether to commit the response to the ChatHistory. That means the following two scenarios are possible:
  * An input has been responded to, and that response is in the chat history
  * An input has been responded to, and that response isn't in the chat history

Would it make sense to similarly permit the webhook implementor to decide whether to mark a message as having been responded to? Basically  make calling `set_response_set` something that wasn't automatic, from within `run_agent`?. 

That feels consistent with the current semantics of `run_agent` not mutating `ChatHistory`

What do you think?